ohttp.go

```diff
@@ -372,10 +370,12 @@ func (c Client) EncapsulateRequest(request []byte) (EncapsulatedRequest, Encapsu
 
 func (c EncapsulatedRequestContext) DecapsulateResponse(response EncapsulatedResponse) ([]byte, error) {
 	// secret = context.Export("message/bhttp response", Nk)
-	secret := c.context.Export(c.responseLabel, c.suite.AEAD.KeySize())
+	_, KDF, AEAD := c.suite.Params()
+
+	secret := c.context.Export(c.responseLabel, AEAD.KeySize())
 
 	// response_nonce = random(max(Nn, Nk)), taken from the encapsualted response
-	responseNonceLen := max(c.suite.AEAD.KeySize(), c.suite.AEAD.NonceSize())
+	responseNonceLen := max(int(AEAD.KeySize()), 12)
```

Comment:
@armfazh, can we get a function in CIRCL that returns the nonce length for an AEAD?

Reply:
https://github.com/cloudflare/circl/pull/424

does this help?

Reply:
Yes, perfect!

Reply:
done